framework/composer.json

```diff
@@ -54,7 +54,6 @@
     "require": {
         "php": ">=5.4.0",
         "ext-mbstring": "*",
-        "ext-mcrypt": "*",
```

Comment:
OpenSSL is a php library. You can specify the requirement as [Composer Platform Packages](https://getcomposer.org/doc/02-libraries.md#platform-packages) specifies with `lib-openssl`. 
Also, why not adding this check in the requirements.php file? I made #7661 and #7663 with this fix.